core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -713,4 +713,9 @@ default boolean isMaterializedView(Session session, QualifiedObjectName viewName
      * Returns a table handle for the specified table name with a specified version
      */
     Optional<TableHandle> getTableHandle(Session session, QualifiedObjectName tableName, Optional<TableVersion> startVersion, Optional<TableVersion> endVersion);
+
+    /**
+     * Returns properties that should not be copied when creating a table like the source table.
+     */
+    Set<String> getTablePropertiesToSkipForCreateTableLike(Session session, CatalogName catalogName);
```

Comment:
I think a nicer approach would be to extend the `PropertyMetadata` class with extra information.

Reply:
Thanks, will give it a try.

Reply:
I agree that would be much cleaner, however this will change the signature of PropertyMetadata and will impact all instantiations everywhere. I also see that there are two instance factories (one with validation and one without), Wouldn't it be beneficial to add a builder class here, if we plan to add more optional properties in the future.

The builder class would have no validation and not to be skipped by default. What do you think?

```
tableProperties = ImmutableList.<PropertyMetadata<?>>builder()
                .add(enumProperty(
                        FILE_FORMAT_PROPERTY,
                        "File format for the table",
                        IcebergFileFormat.class,
                        icebergConfig.getFileFormat(),
                        false))
                .add(new PropertyMetadata<>(
                        PARTITIONING_PROPERTY,
                        "Partition transforms",
                        new ArrayType(VARCHAR),
                        List.class,
                        ImmutableList.of(),
                        false,
                        value -> ((Collection<?>) value).stream()
                                .map(name -> ((String) name).toLowerCase(ENGLISH))
                                .collect(toImmutableList()),
                        value -> value))
                .add(stringProperty(
                        LOCATION_PROPERTY,
                        "File system location URI for the table",
                        null,
                        false))
                .build();
```

Reply:
I would not over-focus on a builder. I think we go with just an extra parameter in constructor and helper `with...` method.
Also, you can add a more generic `flags` field to hold a new bit for "should the property be retained on CREATE...LIKE`; and have a place for further extensions.

Sth like:
```java
public final class PropertyMetadata<T>
{
...
    public enum Flag {
        SKIP_ON_CREATE_LIKE,
    }
...
    private final EnumSet<Flag> flags;
...
    public PropertyMetadata(
            String name,
            String description,
            Type sqlType,
            Class<T> javaType,
            T defaultValue,
            boolean hidden,
            Function<Object, T> decoder,
            Function<T, Object> encoder,
            EnumSet<Flag> flags)
    {...}

    public PropertyMetadata<T> withFlags(Flag... flags)
    {
        new PropertyMetadata<T>(
                name,
                description,
                sqlType,
                javaType,
                defaultValue,
                hidden,
                decoder,
                encoder,
                EnumSet.copyOf(Arrays.asList(flags));
    }
```

Reply:
I think that @losipiuk made a very good suggestion with `PropertyWithMetadata`.
Adding a flag which points out whether the property is to be used or not in `CREATE TABLE ... LIKE` statements is definitely better than having property names scattered in several methods of the connector (the current implementation has `getTablePropertiesToSkipForCreateTableLike` where potentially invalid property names (e.g. `foo`, `bar`) may be specified).

@losipiuk so far we have a enum with one field in the solution you sketched. Do you envision adding more flags (e.g. : `hidden` ) ? In case that the enum stays with one field, I'd suggest using a `boolean` instead for the `useInCreateTableLike` flag.

Reply:
We can migrate `hidden` to be a flag. Then probably we can even skip adding `withFlags` and just refactor constructors we currently have, as effectively we would not add extra complexity (number of fields would remain the same).

Reply:
As most of the properties don't have any flags, i have added constructor factories for those cases to avoid having to create a EnumSet.noneOf(Flag.class) everywhere.